modules/blocks/index.js

```diff
@@ -1,32 +1,51 @@
 export { default as Editable } from './components/editable';
 
 /**
- * Registers a block.
+ * Block settings keyed by block slug.
  *
- * @param  {string} namespace Block grouping unique to package or plugin
- * @param  {string} block     Block name
- * @param  {Object} settings  Block settings
+ * @var {Object} blocks
  */
-export function registerBlock( namespace, block, settings ) {
+const blocks = {};
 
+/**
+ * Validates a block slug and throws an error if it is invalid.
+ *
+ * @param {string} slug Block slug
+ */
+export function validateBlockSlug( slug ) {
+	if ( ! /^[a-z0-9-]+\/[a-z0-9-]+$/.test( slug ) ) {
+		throw new Error(
+			'Block slugs must contain a namespace prefix.  Example:  my-plugin/my-custom-block'
+		)
+	}
 }
 
 /**
- * Returns settings associated with a block.
+ * Registers a block.
  *
- * @param  {string}  namespace Block grouping unique to package or plugin
- * @param  {string}  block     Block name
- * @return {?Object}           Block settings
+ * @param {string} slug     Block slug
+ * @param {Object} settings Block settings
  */
-export function getBlockSettings( namespace, block ) {
+export function registerBlock( slug, settings ) {
+	validateBlockSlug( slug );
```

Comment:
The reason for this design was so that the `validateBlockSlug` function could choose from multiple possible errors to throw, and so that this functionality could be tested independently of registering a block.